diagrams/__init__.py

```diff
@@ -392,4 +394,111 @@ def _load_icon(self):
         return os.path.join(basedir.parent, self._icon_dir, self._icon)
 
 
+class Edge:
+    """Edge represents an edge between two nodes."""
+
+    def __init__(self,
+                 node: "Node" = None,
+                 forward: bool = False,
+                 reverse: bool = False,
+                 label: str = "",
+                 color: str = "",
+                 style: str = ""
+                 ):
+        """Edge represents an edge between two nodes.
+
+        :param node: Parent node.
+        :param forward: Points forward.
+        :param reverse: Points backward.
+        :param label: Edge label.
+        :param color: Edge color.
+        :param style: Edge style.
+        """
+        if node is not None:
+            assert isinstance(node, Node)
+
+        self.node = node
+        self.forward = forward
```

Comment:
You can hold the properties (label, color, style...) in the single `attrs` or `_attrs` dict like `Node` class. And just return the `attrs` dict in `to_dot` method. (I left a comment in that method too)

This is for `Node` class.

```python
self.attrs = {
    "shape": "none",
    "height": str(self._height + padding),
    "image": self._load_icon(),
} if self._icon else {}
```

Reply:
Done

Reply:
You can make it a property even if there are some logics.

```python
    @property
    def attrs(self) -> Dict:
        if self.forward and self.reverse:
            direction = "both"
        elif self.forward:
            direction = "forward"
        elif self.reverse:
            direction = "back"
        else:
            direction = "none"
        return {**self._attrs, "dir": direction}

...

    self.dot.edge(node.hashid, node2.hashid, **edge.attrs)
```

Anything I missed? :)

Reply:
Oh, indeed :) fixed.